### ISSUE TITLE
Feature Request: Add Radioactive Contamination

### ISSUE THREAD

**Issue by ghost** [d3athrow/vgstation13#23621]
<featurerequest>

I'd like to request the addition of radioactive contamination/build up for objects that are exposed to radiation. TG has this feature, and while I know there's a stigma against their code, this seems like a wholly beneficial change by giving players more to do and changing up engineering work a little bit. Right now, radiation never really comes into play unless someone decides to take the shard out for a parade. What this feature would do is essentially keep track of the radiation players/objects are exposed to, and once a certain threshold is passed, turn them into a dangerous source of radiation. Radioactivity would decay over time for people/objects who are no longer exposed to a source, like the shard or singularity, but they would remain dangerous for some amount of time depending on the level of build up. Additionally, to support the new feature, a scanning mode could be added to geiger counters (hold them in hand and click on the object/person you wish to scan) to allow them to detect levels of contamination on a specific object, as well as the ambient radiation they currently detect. 

This could be used to deadly effect by placing an object near a source of radiation and then bringing it back to the rest of the station, where it will slowly (or potentially rapidly depending on how long it was exposed) kill the person who it is given to/picks it up/stands next to it for too long. Imagine you have an assignment to kill the bartender. So you go to the bar and steal a shaker, place it next to the SM core for a few minutes, then return it to him, saying you found it in the maint tunnels. A few minutes later, he starts coughing up blood, and his hair starts falling out, and he has no idea why. 

Pros: 
-Disincentivizes engineer hardsuit rushing (if you're working with the shard itself for extended periods, 75% might not be enough to keep all your stuff rad free). 
-Makes radiation suits more valuable.  
-Gives traitor engineers something else to do besides blowing the engines
-Gives genetics more to do by fixing the genomic damage done by radiation, and doctors more experience curing radiation in general
-!Fun!
-Potentially not much work considering the feature is already implemented in tg and their code base is open source (but I'm not much of a coder so I won't say for sure how much work it is to migrate that stuff)

Cons:
-Potential for griefing via throwing a heavily contaminated object into a high-traffic area. (similar to current shard parades, although less obvious)
-Adds some small amount of complexity to engineering tasks by forcing people to use radiation suits in some circumstances.

**Comment by Irradiation** [comment]
>Disincentivizes engineer hardsuit rushing (if you're working with the shard itself for extended periods, 75% might not be enough to keep all your stuff rad free).

Implying anybody worth their salt uses hardsuits to work/spray (on) the shard instead of radiation suits.


>-Gives genetics more to do by fixing the genomic damage done by radiation, and doctors more experience curing radiation in general

I'm not sure how familiar you are with radiation and how it damages players in this codebase, but just about the only thing you can do to cure radiations is chug Arithrazine, Hyronalin and some Anti-Toxin to deal with the *radiation* aspect. Rad_tick is the itty bitty bit that makes you vomit, bleed and what-not and that part just so happens to only be curable via waiting/chugging a very rare homemade meme drink that also cures it very slowly. And for genomic damage, just throw them into the tubes lmao (Until the day somebody mass-produces Rezadone every round).

**Comment by ghost** [comment]
All fair. I'm not familiar with the way the code works, just the implementation in the game, so I was just kinda spitballing on potential effects.

**Comment by DrSnips** [comment]
>Put a piece(s) of paper next to the [source or radiation like radio transmitter or juke]
Leave it there for a bit
Scoop them up into a bag or clipboard
Put it somewhere
Due to the way radiation interacts with people everyone dies instantly by walking next to it

**Comment by DrSnips** [comment]
Rads are also a medical nightmare because of all of the internal issues it causes and tox damage. 90% of the time you have to clone the person and Id rather have medbay be more than just lmao clone em

**Comment by ghost** [comment]
> > Due to the way radiation interacts with people everyone dies instantly by walking next to it

You could have a cap based on object size/whether it fits into a backpack or not. I'm not a nuclear physicist but my guess is a piece of paper won't absorb as much as a larger object, etc. 

>>Rads are also a medical nightmare because of all of the internal issues it causes and tox damage. 90% of the time you have to clone the person and Id rather have medbay be more than just lmao clone em

Maybe that's an issue with the cloners then. I mean, we have organ printers and stuff, but I don't see them used too often. Admittedly I don't spend much time in medbay but it does seem to me that the cloners are used when other options are available just due to ease of use.

**Comment by DrSnips** [comment]
100 sources of rad is 100 sources of rads. You could load a backpack with 7 cardboard boxes each containing 7 clipboards and each of those containing a max amount of paper. I can think of several other things that you could do something similar with.
The issue here isn't the cloner its rads because of how it JUSTs people. They can even make people literally uncloneable.

**Comment by ghost** [comment]
Yeah. I'm not saying there's no ways to break it, and obviously it requires some amount of refinement, but I still like the general idea. Maybe speaking as someone who doesn't actually know the code that doesn't mean much.

**Comment by SonixApache** [comment]
wyci

**Comment by ghost** [comment]
>>wyci

I'll certainly take a look at the code guides on the wiki. What's involved in porting features from other code bases? I'm sure it's not as simple as a drag and drop, but I can't imagine it's as bad as starting from scratch.

**Comment by DrSnips** [comment]
You get an idea of how to go about coding it most of the time but a lot of the time you're basicly starting from scratch because if Y needs X to work and X isn't a 1 to 1 transfer you're likely going to redo both.
I'm still not for this idea due to how our radiation is.

**Comment by SonixApache** [comment]
Our radiation code is really not built for nor balanced around getting mariecurie'd to death.
The potential for griff and accidental griff is massive.

You'd have to rework all of radiationcode, rework every engine that does anything radiation related, rework all the other sources of radiation (radgun, radstorms, rad meteors, etc), re-design engineering on every map, and rework a bunch of wiki pages -- all while fighting the playerbase who hates major changes.

Interesting concept in theory, ohfuckwhy-tier in practice.

**Comment by ghost** [comment]
Fair enough. I still like the idea and the execution, if done right, is really great (I'll be honest I think the system TG has for rads is a lot more interesting than what we have) but I get that it not the best fit with the way things are coded now, and I don't have the skills to try and rework and balance the whole system myself. Just a suggestion, maybe something that can be looked at in the future. Thanks for your thoughts.

**Comment by tuypo1** [comment]
It sounds cool but potentially way too fucking op
Probably best if it can only slightly iradiate objects.

**Comment by Diavill** [comment]
cant wait to get given the spicy keychain and vomit my insides while being stuck in a cryotube forever

**Comment by FudgePucker** [comment]
The old spicy chunk of graphite

**Comment by FudgePucker** [comment]
Also.
I am going to give my "I dont like change" reply.

So what am I to do as a MoMMI?
Do I have to carefully avoid people until I can get my hands on a geiger counter? Am I to do nothing after spacing an out of control shard? The fuck?

**Comment by ghost** [comment]
>>So what am I to do as a MoMMI?

Well presumably the borgs and mommis have built in radiation shielding considering they already take no damage from working with the core. Realistically if they weren't they'd break down since that level of radiation can easily break circuit boards and such, so you could hand wave it as just them being lead plated or whatever else.

**Comment by tuypo1** [comment]
If you are a mommi what the fuck are you doing spacing shards
Fucking hell we need to be way fucking stricter on the no interference shit.

**Comment by DrSnips** [comment]
As long as there is no one at the shard there's no reason you can't especially if it's something you set up yourself and fucked up.

**Comment by FudgePucker** [comment]
I of course would never space a shard over just using the foam.
